arviz/data/inference_data.py

```diff
@@ -343,13 +363,132 @@ def to_json(self, filename, **kwargs):
         str
             Location of json file
         """
-        idata_dict = _make_json_serializable(self.to_dict())
+        idata_dict = _make_json_serializable(
+            self.to_dict(groups=groups, filter_groups=filter_groups)
+        )
 
         with open(filename, "w") as file:
             json.dump(idata_dict, file, **kwargs)
 
         return filename
 
+    def to_dataframe(
+        self,
+        groups=None,
+        filter_groups=None,
+        include_coords=True,
+        include_index=True,
+        index_origin=None,
+    ):
+        """Convert InferenceData to a pandas DataFrame following xarray naming conventions.
+
+        This returns dataframe in a "wide" -format, where each item in ndimensional array is
+        unpacked. To access "tidy" -format, use xarray functionality found for each dataset.
+
+        In case of a multiple groups, function adds a group identification to the var name.
+
+        Data groups ("observed_data", "constant_data", "predictions_constant_data") are
+        skipped implicitly.
+
+        Raises TypeError if no valid groups are found.
+
+        Parameters
+        ----------
+        groups: str or list of str, optional
+            Groups where the transformation is to be applied. Can either be group names
+            or metagroup names.
+        filter_groups: {None, "like", "regex"}, optional, default=None
+            If `None` (default), interpret groups as the real group or metagroup names.
+            If "like", interpret groups as substrings of the real group or metagroup names.
+            If "regex", interpret groups as regular expressions on the real group or
+            metagroup names. A la `pandas.filter`.
+        include_coords: bool
+            Add coordinate values to column name (tuple).
+        include_index: bool
+            Add index information for multidimensional arrays.
+        index_origin: {0, 1}, optional
+            Starting index  for multidimensional objects. 0- or 1-based.
+            Defaults to rcParams["data.index_origin"].
+
+        Returns
+        -------
+        pandas.DataFrame
+            A pandas DataFrame containing all selected groups of InferenceData object.
+        """
+        # pylint: disable=too-many-nested-blocks
+        if not include_coords and not include_index:
+            raise TypeError("Both include_coords and include_index can not be False.")
+        if index_origin is None:
+            index_origin = rcParams["data.index_origin"]
+        if index_origin not in [0, 1]:
+            raise TypeError("index_origin must be 0 or 1, saw {}".format(index_origin))
+
+        group_names = list(
+            filter(lambda x: "data" not in x, self._group_names(groups, filter_groups))
+        )
+
+        if not group_names:
+            raise TypeError("No valid groups found: {}".format(groups))
```

Comment:
Is this logic ok?

posterior + observed_data --> df from posterior

observed_data -> TypeError

Reply:
I think it is ok, given that it fits our goals and we document it. My only question not really knowing yet how panel works is if ppc checks will be possible with it if observed_data is not in the dataframe

Reply:
Panel works with array too. My gist was just short cutting some code with data frame.

Reply:
Oh, cool!